akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/internal/ShardedDaemonProcessImpl.scala

```diff
@@ -130,7 +39,15 @@ private[akka] final class ShardedDaemonProcessImpl(system: ActorSystem[_])
     extends javadsl.ShardedDaemonProcess
     with scaladsl.ShardedDaemonProcess {
 
-  import ShardedDaemonProcessImpl._
+  import ShardedDaemonProcessId._
+  import ShardedDaemonProcessState.verifyRevisionBeforeStarting
+
+  private case class ShardedDaemonProcessContextImpl(
+      processNumber: Int,
+      totalProcesses: Int,
+      name: String,
+      revision: Long)
+      extends ShardedDaemonProcessContext
 
   def init[T](name: String, numberOfInstances: Int, behaviorFactory: Int => Behavior[T])(
```

Comment:
we are missing a corresponding `initWithContext` for this one. good to have all 3 to make the "migration" seamless.

Reply:
I tried that but got compiler errors because of ambiguous overloads in some way I didn't figure out a way around, so backed out ot of it completely. Can try giving it another shot.

Reply:
Done in [65acbec](https://github.com/akka/akka/pull/31844/commits/65acbec08bb9fd4ad4084f7e7726e9373338616c) and noticed that I left the old ddata state timeout config in place, so removed that in [ff0ae04](https://github.com/akka/akka/pull/31844/commits/ff0ae043e3bd75c53772a0e370c1a1dd7e9beb5d)